contract-tests/test_v3_api.py

```diff
@@ -0,0 +1,240 @@
+import json
+import jsonschema
+import magic
+import pytest
+
+from os.path import join, dirname
+from random import randint
+
+
+def _load_json_schema(filename):
+    relative_path = join("schemas", filename)
+    absolute_path = join(dirname(__file__), relative_path)
+
+    with open(absolute_path) as schema_file:
+        return json.loads(schema_file.read())
+
+
+def assert_valid_schema(data, schema_file):
+    schema = _load_json_schema(schema_file)
+    return jsonschema.validate(data, schema)
+
+
+def test_v3_actions(conf, requests_session):
+    # Get a random action and validate it
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("No list of v3 actions found")
+
+    key = randint(0, len(data["results"]) - 1)
+    action_id = data["results"][key]["id"]
+
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/{}/".format(action_id))
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("Could not find a v3 action with the ID {}".format(action_id))
+
+    assert r.status_code == 200
+    assert_valid_schema(data, "v3.action.schema")
+
+
+def test_v3_approval_request(conf, requests_session):
+    r = requests_session.get(conf.getoption("server") + "/api/v3/approval_request/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0 or len(data["results"]) == 0:
+        pytest.skip("No list of v3 approval requests found")
+
+    assert r.status_code == 200
+
+
+def test_v3_approval_request_with_id(conf, requests_session):

```

Comment:
Is there a reason you split this into two tests, but made the action version a single test?

Reply:
No reason, I can just split them into two different tests